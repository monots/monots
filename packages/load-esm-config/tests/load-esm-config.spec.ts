import { createSetupFixtures } from '@monots/test';
import { afterAll, expect, test } from 'vitest';

import { loadEsmConfig } from '../';

export const setupFixtures = createSetupFixtures({ context: {}, fileUrl: import.meta.url });

afterAll(async () => {
  await setupFixtures.cleanup();
});

test.each([
  ['.js', 'simple'],
  ['.mjs', 'simple2'],
  ['.ts', 'simple3'],
  ['.mts', 'simple4'],
])('can load simple `%s` config in commonjs environment', async (extension, name) => {
  const { context } = await setupFixtures('cjs');
  const { config, path } = await loadEsmConfig({ name, cwd: context.cwd });

  expect(path.endsWith(`${name}.config${extension}`)).toBe(true);
  expect(config).toMatchSnapshot();
});

test.each([
  ['.js', 'simple'],
  ['.cjs', 'simple2'],
  ['.ts', 'simple3'],
  ['.cts', 'simple4'],
])('can load simple `%s` config in esm environment', async (extension, name) => {
  const { context } = await setupFixtures('esm');
  const { config, path } = await loadEsmConfig({ name, cwd: context.cwd });

  expect(path.endsWith(`${name}.config${extension}`)).toBe(true);
  expect(config).toMatchSnapshot();
});
