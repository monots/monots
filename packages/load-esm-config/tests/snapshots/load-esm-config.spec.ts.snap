// Vitest Snapshot v1

exports[`can load simple \`.cjs\` config in esm environment 1`] = `
{
  "filename": "simple2.config.cjs",
  "option": "simple2",
}
`;

exports[`can load simple \`.cts\` config in esm environment 1`] = `
{
  "filename": "simple4.config.cts",
  "option": "simple4",
}
`;

exports[`can load simple \`.js\` config in commonjs environment 1`] = `
{
  "filename": "simple.config.js",
  "option": "simple",
}
`;

exports[`can load simple \`.js\` config in esm environment 1`] = `
{
  "filename": "simple.config.js",
  "option": "simple",
}
`;

exports[`can load simple \`.mjs\` config in commonjs environment 1`] = `
{
  "filename": "simple2.config.mjs",
  "option": "simple2",
}
`;

exports[`can load simple \`.mts\` config in commonjs environment 1`] = `
{
  "filename": "simple4.config.mts.js",
  "option": "simple4",
}
`;

exports[`can load simple \`.ts\` config in commonjs environment 1`] = `
{
  "filename": "simple3.config.ts",
  "option": "simple3",
}
`;

exports[`can load simple \`.ts\` config in esm environment 1`] = `
{
  "filename": "simple3.config.ts",
  "option": "simple3",
}
`;
